### PR TITLE
#3892 Contracts: Allow possibility to copy the prices when extending 

### PR DESCRIPTION
**Pull request by cristinamghita**
#3892 Contracts: Allow possibility to copy the prices when extending

### CODE REVIEW

File: de.metas.contracts/src/main/sql/postgresql/system/50-de.metas.contracts/5491360_sys_gh3892_AddIsCalculatePrice.sql [hunk 1]
```diff
@@ -0,0 +1,42 @@
+-- 2018-04-19T18:01:52.240
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+INSERT INTO AD_Element (AD_Client_ID,AD_Element_ID,AD_Org_ID,ColumnName,Created,CreatedBy,Description,EntityType,Help,IsActive,Name,PrintName,Updated,UpdatedBy) VALUES (0,543964,0,'IsCalculatePrice',TO_TIMESTAMP('2018-04-19 18:01:51','YYYY-MM-DD HH24:MI:SS'),100,'','de.metas.contracts','','Y','Calculate Price','Calculate Price',TO_TIMESTAMP('2018-04-19 18:01:51','YYYY-MM-DD HH24:MI:SS'),100)
```
Comment (metas-ts) [inline de.metas.contracts/src/main/sql/postgresql/system/50-de.metas.contracts/5491360_sys_gh3892_AddIsCalculatePrice.sql]: a Y/N column `IsCalculatePrice` and the text "CAlculate Price" doesn't give the user enough info.
E.g. what should a user expect if  they set IsCalculatePrice=N ? a prompt to set the price manually? everything fille with zeros?
Adding more documentation would be one solution, but i'm not sure how the user can access it.
so omho the best solution is to have a list with two options:
* column/field onFlatrateTermExtend (or similar, please alligne with existing column names)
* values: "copy prices from predecessor" and "calculate from pricing data"

Comment (metas-ts) [reply de.metas.contracts/src/main/sql/postgresql/system/50-de.metas.contracts/5491360_sys_gh3892_AddIsCalculatePrice.sql]: btw, please chag you the test failure

Comment (metas-ts) [reply de.metas.contracts/src/main/sql/postgresql/system/50-de.metas.contracts/5491360_sys_gh3892_AddIsCalculatePrice.sql]: Another thing: do we *really* want to have this option in C_Flatrate_Condition?
The issue sais "Contracts: Allow possibility to copy the prices when extending"
* At which stage shall this be allowed (C_Flatrate_Condition is a rather early stage), and 
* in which cases shall the pricese be calculated and in which cases shall they be copied (with this implementation, they are *always* calculated or *always* copied)

Comment (metasnw) [reply de.metas.contracts/src/main/sql/postgresql/system/50-de.metas.contracts/5491360_sys_gh3892_AddIsCalculatePrice.sql]: results from our talk:
1. Checbox => List box with explicit names
1. Fix tests
3. we keep the option in flatrate conditions

**Pull request merged**